### PR TITLE
Better regular expression for verifying vote in verify vote - Close #627

### LINKED ISSUE #627: Better regular expression for vote in logic/vote -> verifyVote function
In verifyVote function, we check the format of vote using `if (!/[-+]{1}[0-9a-z]{64}/.test(vote)) ` condition, we should update regular expression to only pass for hexadecimal strings.

### PR DESCRIPTION
**Pull request by SargeKhan**
### What was the problem?
Vote regular expression was passing on non-hex vote string.
### How did I fix it?
Updated the regular expression to fail on non-hex vote string.
### How to test it?
Run the unit tests
### Review checklist

* The PR resolves #627 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: logic/vote.js [hunk 1]
```diff
@@ -227,14 +227,10 @@ Vote.prototype.verifyVote = function(vote, cb) {
 		return setImmediate(cb, 'Invalid vote type');
 	}
 
-	if (!/[-+]{1}[0-9a-z]{64}/.test(vote)) {
+	if (!/^[-+]{1}[0-9a-f]{64}$/.test(vote)) {
```
Comment (willclarktech) [inline logic/vote.js]: `{1}` is redundant here

Comment (SargeKhan) [reply logic/vote.js]: I think it makes the regex more descriptive. What do you think?

File: test/unit/logic/vote.js [hunk 2]
```diff
@@ -378,7 +378,7 @@ describe('vote', () => {
 	});
 
 	describe('verifyVote', () => {
-		it('should throw if vote is of invalid length', done => {
+		it('should return error if vote is of invalid length', done => {
```
Comment (ManuGowda) [inline test/unit/logic/vote.js]: The description and expected error doesn't match, it should be `should return error if vote is of invalid format`

Comment (SargeKhan) [reply test/unit/logic/vote.js]: the test description doesn't say what the error will be. The test description specifies what the kind of input we pass to the function.

**Pull request merged**